network-protection/network-protection-impl/src/test/java/com/duckduckgo/networkprotection/impl/reconnect/NetpVpnConnectivityLossListenerPluginTest.kt

```diff
@@ -83,19 +96,33 @@ class NetpVpnConnectivityLossListenerPluginTest {
     }
 
     @Test
-    fun whenOnVpnConnectivityLossCalledTwiceThenGiveUpRecoveringAndUnregisterNetp() = runTest {
+    fun whenOnVpnConnectivityLossAndReachedMaxAttemptsThenGiveUpRecoveringAndUnregisterNetp() = runTest {
         whenever(vpnFeaturesRegistry.isFeatureRegistered(NetPVpnFeature.NETP_VPN)).thenReturn(true)
-        repository.reconnectAttemptCount = 1
+        whenever(stopWatch.elapsedRealtime()).thenReturn(TimeUnit.MINUTES.toMillis(1) + 1)
+        testee.onVpnConnectivityLoss(coroutineRule.testScope)
```

Comment:
I guess we had to call twice here because the first call on onConnectivityLoss always guarantees to initiate recovery.

Reply:
Yeah, that's exactly why